# Regression:
* Linear Regression emphasis relation betweeen 2 variables and tries to fit relation through a line. Y = mX + C, X is the independent variable and Y is the dependent variable and C is the bias normalizer.
* Scatter Plot and Corelation coefficients are the popular metrics to determine relation beween 2 variables/features.
* Scatter plot would have x axis as independent and y axis as the one where would like to see the pattern of the target/dependent variable. It also indicates the direction of dependency.
* Corelation coefficients are the values between -1 to 1. where negative indicates indirect co relation and values closer to -1 and 1 indicates the max strength of the co relation.
* Least squarred error(popular method) is the way through which a best fit line is chosen.

In [1]:
import pandas as pd
import statsmodels.api as sm

In [2]:
df = pd.read_csv("train.csv")
df.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea    ...    MoSold YrSold SaleType SaleCondition SalePrice
0   1          60       RL         65.0     8450    ...         2   2008       WD        Normal    208500
1   2          20       RL         80.0     9600    ...         5   2007       WD        Normal    181500
2   3          60       RL         68.0    11250    ...         9   2008       WD        Normal    223500
3   4          70       RL         60.0     9550    ...         2   2006       WD       Abnorml    140000
4   5          60       RL         84.0    14260    ...        12   2008       WD        Normal    250000

[5 rows x 81 columns]

In [3]:
df[["GrLivArea","SalePrice"]].head()

GrLivArea  SalePrice
0       1710     208500
1       1262     181500
2       1786     223500
3       1717     140000
4       2198     250000

In [6]:
#adding intercept
df['intercept'] = 1

lm = sm.OLS(df['SalePrice'],df[['intercept','GrLivArea']])
results = lm.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.502
Model:                            OLS   Adj. R-squared:                  0.502
Method:                 Least Squares   F-statistic:                     1471.
Date:                Mon, 25 Mar 2019   Prob (F-statistic):          4.52e-223
Time:                        19:43:29   Log-Likelihood:                -18035.
No. Observations:                1460   AIC:                         3.607e+04
Df Residuals:                    1458   BIC:                         3.608e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
intercept   1.857e+04   4480.755      4.144      0.000    9779.612    2.74e+04
GrLivArea    107.1304      2.794     38.348      0.000     101.650     112.610
==============================================================================
Omnibus:                      261.166   Durbin-Watson:                   2.025
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             3432.287
Skew:                           0.410   Prob(JB):                         0.00
Kurtosis:                      10.467   Cond. No.                     4.90e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.9e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**Coef**
* intecept values indicate that when area is 0 the price of the house is 18570
* for one unit increase in area, sale price would increase by 107 dollar price.

**P>[t]**
* 0.000 (low value) indicates that the variable is important for SalePrice prediction

**R2 Squred**
* co relation between the variables (SalePrice and Area)